### PR TITLE
[New] Show popup

### PR DESCRIPTION
**Pull request by des12437**
## Description

This PR implements `Show popup` in `Display information` category.
URL to README: [URL](https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/des12437/Show-popup-sample/Shared/Samples/Show%20popup/README.md)

## Linked Issue(s)

- `swift/issues/3579`

## How To Test
- Tap any feature on the map to see a popup with the attributes of the feature listed.
- Tap the close icon or swipe down on the popup view to dismiss the popup.

### CODE REVIEW

File: Shared/Samples/Show popup/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Show popup
+
+Show a predefined popup from a web map.
+
+![Show popup screenshot](show-popup.png)
```
Comment (dfeinzimer) [inline Shared/Samples/Show popup/README.md:5]: We may want to downsize this image a bit. It's smaller in the other READMEs and overpowers the rest of the content a bit. I _think_ the others use a width of 276.

File: Shared/Samples/Show popup/ShowPopupView.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import ArcGIS
+import ArcGISToolkit
+
+struct ShowPopupView: View {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    private class Model: ObservableObject {
+        /// A map of reported incidents in San Francisco.
+        let map = Map(url: URL(string: "https://arcgisruntime.maps.arcgis.com/home/item.html?id=fb788308ea2e4d8682b9c05ef641f273")!)!
+    }
```
Comment (dfeinzimer) [inline Shared/Samples/Show popup/ShowPopupView.swift:25]: Similar to the work in #138, I think we can do without the model for this sample and just store the map as a `@State` property instead.

File: Samples.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1148,21 +1173,9 @@
...
-/* Begin XCRemoteSwiftPackageReference section */
-		00C43AEB2947DC350099AE34 /* XCRemoteSwiftPackageReference "arcgis-maps-sdk-swift-toolkit" */ = {
-			isa = XCRemoteSwiftPackageReference;
-			repositoryURL = "https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/";
-			requirement = {
-				kind = upToNextMinorVersion;
-				minimumVersion = 200.0.0-beta;
-			};
-		};
-/* End XCRemoteSwiftPackageReference section */
```
Comment (dfeinzimer) [inline Samples.xcodeproj/project.pbxproj:1160]: It looks like the references to the toolkit got changed. Looks like it may have all happened in 7c1a992d392e2d14c3972bc8fbf48649d9a85e73. I'd look into reverting that commit and re-adding the tag changes only.

**Review by dfeinzimer — DISMISSED** [review]

File: Shared/Samples/Show popup/ShowPopupView.swift [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                          })
+                        .cancellationToNil()
+                    else {
+                        return
+                    }
+                    
+                    self.identifyScreenPoint = nil
+                    self.popup = try? identifyResult.get().first?.popups.first
+                    self.showPopup = self.popup != nil
+                }
```
Comment (yo1995) [inline Shared/Samples/Show popup/ShowPopupView.swift:58]:
```swift
.task(id: identifyScreenPoint) {
    guard let identifyScreenPoint = identifyScreenPoint,
          let identifyResult = try? await proxy.identifyLayers(
              screenPoint: identifyScreenPoint,
              tolerance: 12,
              returnPopupsOnly: false
          ),
          let firstPopup = identifyResult.first?.popups.first
    else { return }
    
    self.popup = firstPopup
    self.showPopup = true
}
```

A few things…

- I don't think there is a UI to cancel the task, so no need for `cancellationToNil`
- Show the floating panel only when there is a popup.

File: Shared/Samples/Show popup/ShowPopupView.swift [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+import ArcGISToolkit
+
+struct ShowPopupView: View {
+    /// A map of reported incidents in San Francisco.
+    @State private var map = Map(url: URL(string: "https://arcgisruntime.maps.arcgis.com/home/item.html?id=fb788308ea2e4d8682b9c05ef641f273")!)!
```
Comment (yo1995) [inline Shared/Samples/Show popup/ShowPopupView.swift:21]: In the samples we prefer using portal item over URLs

- No need to unwrap URL
- Make it easier to search for/pattern match all the portal item IDs we use in the samples
- Code looks shorter

```swift
    @State private var map = Map(
        item: PortalItem(
            portal: .arcGISOnline(connection: .anonymous),
            id: .incidentsInSanFrancisco
        )
    )

// at the bottom
private extension PortalItem.ID {
    /// The ID used in the "Incidents in San Francisco" portal item.
    static var incidentsInSanFrancisco: Self { Self("fb788308ea2e4d8682b9c05ef641f273")! }
}
```

File: Shared/Samples/Show popup/README.metadata.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    ],
+    "keywords": [
+            "feature",
+            "feature layer",
+            "popup",
+            "web map",
+            "IdentifyLayerResult",
+            "Map",
+            "PopupView"
+        ],
```
Comment (yo1995) [inline Shared/Samples/Show popup/README.metadata.json:16]: Please re-indent the json

File: Shared/Samples/Show popup/README.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+Many web maps contain predefined popups which are used to display the attributes associated with each feature layer in the map, such as hiking trails, land values, or unemployment rates. You can display text, attachments, images, charts, and web links. Rather than creating new popups to display information, you can easily access and display the predefined popups.
+
+## How to use the sample
+
+Tap on the features to prompt a popup that displays information about the feature.
+
+## How it works
+
+1. Create a `Map` using a URL.
```
Comment (yo1995) [inline Shared/Samples/Show popup/README.md:17]: If the suggestion is applied above, also adjust phrasing here.

File: Shared/Samples/Show popup/README.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+2. Create a `MapView` with the `Map`.
+3. Use the `GeoViewProxy.identifyLayers(_:screenPoint:tolerance:returnPopupsOnly:maximumResultsPerLayer:)` method to identify the top-most features.
+4. Create a `PopupView` with the result's `popups`.
```
Comment (yo1995) [inline Shared/Samples/Show popup/README.md:20]:
```suggestion
3. Use the `GeoViewProxy.identifyLayers(_:screenPoint:tolerance:returnPopupsOnly:maximumResultsPerLayer:)` method to identify the top-most feature.
4. Create a `PopupView` with the result's first popup.
```
In the past the `AGSPopupsViewController` supports displaying multiple popups, but that support is not added yet. @mhdostal Can you help add a link to the discussion for multiple popup? I saw it somewhere but cannot recall where. Thanks.

File: Shared/Samples/Show popup/README.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+## Relevant API
+
+* IdentifyLayerResult
+* Map
+* PopupView
+
+## About the data
+
+This sample uses a [feature layer](https://sampleserver6.arcgisonline.com/arcgis/rest/services/SF311/FeatureServer/0) that displays reported incidents in San Francisco.
```
Comment (yo1995) [inline Shared/Samples/Show popup/README.md:30]:
```suggestion
This sample uses a [web map](https://www.arcgis.com/home/item.html?id=fb788308ea2e4d8682b9c05ef641f273) that displays reported incidents in San Francisco.
```

**Review by yo1995 — COMMENTED** [review]
Please also fix the merge conflicts.

File: Shared/Samples/Show popup/README.md (hunk 8 above)
Comment (mhdostal) [reply Shared/Samples/Show popup/README.md:20]: There is currently no planned support for multiple popups.  It was decided that clients could easily implement that themselves if desired.  I can't find an online discussion about it; I just remember a verbal conversation that it wasn't needed.

Comment (yo1995) [reply Shared/Samples/Show popup/README.md:20]: Thanks for the clarification!

File: Samples.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -514,12 +516,12 @@
 		00E5400B27F3CCA100CF66D5 /* Shared */ = {
 			isa = PBXGroup;
 			children = (
-				0074ABAF281742420037244A /* Supporting Files */,
```
Comment (yo1995) [inline Samples.xcodeproj/project.pbxproj:517]: Please revert this change. The upper level folders don't need to be sorted, only the samples folders need to.

File: Shared/Samples/Show popup/ShowPopupView.swift [hunk 11]
```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 Esri
```
Comment (yo1995) [inline Shared/Samples/Show popup/ShowPopupView.swift:1]:
```suggestion
// Copyright 2023 Esri
```

**Review by yo1995 — DISMISSED** [review]
Feel free to mark conversations as resolved yourself.

**Review by yo1995 — APPROVED** [review]

**Review by dfeinzimer — APPROVED** [review]
